arduino-ide-extension/src/browser/serial/monitor/serial-monitor-send-input.tsx

```diff
@@ -6,6 +6,75 @@ import { BoardsServiceProvider } from '../../boards/boards-service-provider';
 import { MonitorModel } from '../../monitor-model';
 import { Unknown } from '../../../common/nls';
 
+class RingList {
+  protected ring: string[];
+  protected size: number;
+  protected begin: number;
+  protected index: number;
+  protected end: number;
+
+  constructor(size: number = 100) {
+    this.Init = this.Init.bind(this);
+    this.Push = this.Push.bind(this);
+    this.Prev = this.Prev.bind(this);
+    this.Next = this.Next.bind(this);
+    this.Init(size);
+  }
+
+  public Init(size: number = 100)
+  {
+    this.ring = [];
+    this.size = (size > 0) ? size : 1;
+    this.begin = 0;
+    this.index = 0;
+    this.end = -1;
+  }
+
+  public Push(val: string): number {
+    this.end++;
+    if (this.ring.length >= this.size)
+    {
+      if (this.end >= this.size) 
+        this.end = 0;
+      if (this.end === this.begin)
+      {
+        this.begin++;
+        if (this.begin >= this.size) 
+          this.begin = 0;
+      }
+    }
+    this.ring[this.end] = val;
+    this.index = this.end;
+
+    return this.index;
+  }
+
+  public Prev(): string {
+    if (this.ring.length < 1) {
+      return "";
```

Comment:
Please use single-quotes for strings. `'` is OK, `"` is only OK if you want to escape `'`.

Examples:
`'OK'`, ✅ 
`"NOK"`, 🔥 
`"If you escape single quotes: It's OK."`. ✅

Reply:
Done